packages/apptive_grid_core/lib/network/apptive_grid_client.dart

```diff
@@ -52,12 +52,18 @@ class ApptiveGridClient {
 
   /// Headers that are used for multiple Calls
   @visibleForTesting
-  Map<String, String> get headers => (<String, String?>{
+  Map<String, String> get defaultHeaders => (<String, String?>{
         HttpHeaders.authorizationHeader: _authenticator.header,
         HttpHeaders.contentTypeHeader: ContentType.json,
       }..removeWhere((key, value) => value == null))
           .map((key, value) => MapEntry(key, value!));
 
+  Map<String, String> defaultHeader(Map<String, String> customHeader) {
+    var newHeader = defaultHeaders;
+    newHeader.addAll(customHeader);
```

Comment:
Also I think we might want to turn around the order that we add the default headers to any headers that are passed in by the user. That way we can make sure that users can't override any necessary headers like authentication or accept headers

Reply:
I thought this is exactly there for the user to have the option to do that if needed. I would say, if the user overrides necessary fields, that is on them